test/integration/relations.test.js

```diff
@@ -401,123 +494,261 @@ describe('Relations', function () {
+                const query = makeQuery(mongoJSON);
 
-                // Check any intermediate values
-                console.log(query.toQuery());
+                return query
+                    .select()
+                    .then((result) => {
+                        result.should.be.an.Array().with.lengthOf(1);
+                        result[0].title.should.equal('When She Loved Me');
+                    });
+            });
 
-                // Perform the query against the DB
-                return query.select()
+            it('tags.slug equals "animal" and posts_tags.sort_order is 0 and featured is false', function () {
+                // where primary tag is "animal"
+                const mongoJSON = {
+                    $and: [
+                        {
+                            $and: [
+                                {
+                                    'tags.slug': 'animal'
+                                },
+                                {
+                                    'posts_tags.sort_order': 0
+                                }
+                            ]
+                        },
+                        {
+                            featured: false
+                        }
+                    ]
+                };
+
+                const query = makeQuery(mongoJSON);
+
+                return query
+                    .select()
                     .then((result) => {
-                        console.log(result);
+                        result.should.be.an.Array().with.lengthOf(1);
+                        result[0].title.should.equal('The Bare Necessities');
+                    });
+            });
 
-                        // NOTE: make sure to count in posts with no tags
-                        //       do not count in posts with multiple tags containing one of the excluded tags
+            it('tags.slug NOT equal "classic" and posts_tags.sort_order is 0 and featured is true', function () {
+                const mongoJSON = {
+                    $and: [
+                        {
+                            $and: [
+                                {
+                                    'tags.slug': {
+                                        $ne: 'classic'
+                                    }
+                                },
+                                {
+                                    'posts_tags.sort_order': 0
+                                }
+                            ]
+                        },
+                        {
+                            featured: true
+                        }
+                    ]
+                };
+
+                const query = makeQuery(mongoJSON);
+
+                return query
+                    .select()
+                    .then((result) => {
+                        // TODO: should this include tags with no tags? the filter is about primary tag != 'classic' so they should count?
```

Comment:
needs clarification :man_shrugging:  but seems correct

Reply:
Tag with no tags?

Reply:
tags with no tags!!! and posts without posts!!!